engine/src/main/scala/cromwell/engine/workflow/CallCachingBlacklistManager.scala

```diff
@@ -0,0 +1,132 @@
+package cromwell.engine.workflow
+
+import akka.event.LoggingAdapter
+import com.google.common.cache.{CacheBuilder, CacheLoader, LoadingCache}
+import com.typesafe.config.Config
+import cromwell.backend.standard.callcaching.{BlacklistCache, GroupingBlacklistCache, RootWorkflowBlacklistCache}
+import cromwell.core.CacheConfig
+import mouse.boolean._
+import net.ceedubs.ficus.Ficus._
+
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+object CallCachingBlacklistManager {
+  object Defaults {
+    object Groupings {
+      val Concurrency = 1000
+      val Size = 1000L
+      val Ttl = 2 hours
+    }
+    object Hits {
+      val Concurrency = 1000
+      val Size = 20000L
+      val Ttl = 1 hour
+    }
+    object Buckets {
+      val Concurrency = 1000
+      val Size = 1000L
+      val Ttl = 1 hour
+    }
+  }
+}
+
+class CallCachingBlacklistManager(rootConfig: Config, logger: LoggingAdapter) {
+
+  // Defined if "call-caching.blacklist-cache.enabled = true".
+  private val blacklistCacheConfig: Option[Unit] =
+    rootConfig.getOrElse("call-caching.blacklist-cache.enabled", false).option(())
```

Comment:
Any reason this has to be an `Option[Unit]` rather than a boolean? Was it originally going to be an `Option` containing the config iff it existed?

Reply:
`Option[Boolean]` has three possible values which is kind of confusing when there are really only two. The point of this is just to short-circuit the dependent forcomps if the blacklist cache is not enabled.